### PR TITLE
Annotate System.Security.AccessControl for nullable

### PR DESCRIPTION
**Pull request by buyaa-n**
Contributes to https://github.com/dotnet/corefx/issues/40623

### CODE REVIEW

**Review by eiriktsarpalis — APPROVED** [review]
LGMT, thanks!

File: src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/ACE.cs [hunk 1]
```diff
@@ -328,16 +323,7 @@ public static GenericAce CreateFromBinaryForm(byte[] binaryForm, int offset)
...
-                int accessMask;
-                SecurityIdentifier sid;
-                ObjectAceFlags objectFlags;
-                Guid objectAceType;
-                Guid inheritedObjectAceType;
-                bool isCallback;
-                byte[] opaque;
-
-                if (true == ObjectAce.ParseBinaryForm(binaryForm, offset, out qualifier, out accessMask, out sid, out objectFlags, out objectAceType, out inheritedObjectAceType, out isCallback, out opaque))
+                if (ObjectAce.ParseBinaryForm(binaryForm, offset, out AceQualifier qualifier, out int accessMask, out SecurityIdentifier? sid, out ObjectAceFlags objectFlags, out Guid objectAceType, out Guid inheritedObjectAceType, out bool isCallback, out byte[]? opaque))
```
Comment (bartonjs) [inline src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/ACE.cs]: Moving these variable declarations inline has taken a long line and made it a very long line.  Consider chopping, or just undo moving the variable declarations.

File: src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/ACL.cs [hunk 2]
```diff
@@ -68,7 +68,7 @@ object IEnumerator.Current
 
         public GenericAce Current
         {
-            get { return ((IEnumerator)this).Current as GenericAce; }
+            get { return (((IEnumerator)this).Current as GenericAce)!; }
```
Comment (bartonjs) [inline src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/ACL.cs]: Why did you use ! here, but a hard cast on GetEnumerator?  Seems like changing as-cast to hard-cast would make this look nicer.

File: src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/ACE.cs (hunk 1 above)
Comment (buyaa-n) [reply src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/ACE.cs]: Yeah, will chop the line

File: src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/ACL.cs (hunk 2 above)
Comment (buyaa-n) [reply src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/ACL.cs]: Well this is already has a cast and adding another cast might look confusing, but am ok with hard cast

**Review by bartonjs — APPROVED** [review]

**Comment by buyaa-n** [comment]
The left CI failure is unrelated to this PR, merging

**Pull request merged**